ksqldb-functional-tests/src/test/resources/query-validation-tests/joins.json

```diff
@@ -1907,37 +1907,19 @@
       }
     },
     {
-      "name": "matching time-windowed",
+      "name": "matching time-windowed join with different windows fails",
```

Comment:
Why does this test fail with this change? I don't see the connection to key serde features.

Reply:
because previously we were checking just that the `FormatInfo` matched, which didn't include window information. `KeyFormat` does

Reply:
Makes sense. We should call it out as a breaking change, though.

Reply:
Thanks for adding the breaking change description to the PR! Friendly reminder to copy it into the commit message when this gets merged, in the format `BREAKING CHANGE: <text here>` so the auto-changelog-generator picks it up :)